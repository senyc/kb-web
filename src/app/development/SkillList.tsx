
interface ListItem {
  heading: string;
  experienceWith: string;
}

interface SkillListProps {
  skills: ListItem[];
};

export default function SkillList({ skills }: SkillListProps) {
  return (
    <ul>
      <li>
        <div className='flex w-96 flex-row'>
          <h2 className='text-xl font-bold'>
            Tool/Language
          </h2>
          <h3 className='grow text-right text-xl font-bold'>
            Experience
          </h3>
        </div>
          <div className='mb-8 mt-1 h-0 border opacity-30'> </div>
      </li>
      {skills.map(skill => (
        <li key={skill.heading} className='mb-8'>
          <div className='flex w-96 flex-row'>
            <h2 className='text-lg font-bold'>
              {skill.heading}
            </h2>
            <h3 className='grow text-right text-lg italic'>
              {skill.experienceWith}
            </h3>
          </div>
        </li>
      ))}
    </ul>
  );
}
