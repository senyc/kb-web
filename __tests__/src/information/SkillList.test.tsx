import { create } from 'react-test-renderer';

import SkillList from '@components/information/SkillList';

describe('SkillList elements match recorded snapshots', () => {
  describe('SkillList', () => {
    it('renders items', () => {
      const tree = create(<SkillList skills={[{heading: 'one', experienceWith:'1 Year'}]} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
