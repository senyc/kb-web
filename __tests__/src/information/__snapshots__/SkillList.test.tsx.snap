// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillList elements match recorded snapshots SkillList renders items 1`] = `
<ul
  className="flex flex-col"
>
  <li
    className="border-b-text mb-2 flex flex-row border-b-2 pb-1"
  >
    <h2
      className="font-bold sm:text-xl"
    >
      Tool/Language
    </h2>
    <h3
      className="grow text-right font-bold sm:text-xl"
    >
      Experience
    </h3>
  </li>
  <li
    className="mb-6 flex flex-row"
  >
    <h2
      className="text-lg font-bold"
    >
      one
    </h2>
    <h3
      className="grow text-right text-lg italic"
    >
      1 Year
    </h3>
  </li>
</ul>
`;
